pkg/controller/versions/strategy_v2_3.go

```diff
@@ -199,6 +201,41 @@ func (v *versionStrategyV2_3) validateAddons(spec *v2.ControlPlaneSpec, allError
 	return allErrors
 }
 
+func (v *versionStrategyV2_3) validateServiceMeshExtensionsRemoved(ctx context.Context, cl client.Client, smcp metav1.Object) error {
+	logger := logf.Log.WithName("smcp-validator-2.3")
+
+	serviceMeshExtensions := &apiv1.ServiceMeshExtensionList{}
+	if err := cl.List(ctx, serviceMeshExtensions); err != nil {
+		if !errors.IsNotFound(err) && !meta.IsNoMatchError(err) {
+			// log error, but don't fail validation- this is a convenience check
+			logger.Error(err, "failed to list ServiceMeshExtensions")
+			return nil
+		}
+	}
+	if len(serviceMeshExtensions.Items) > 0 {
+		smmr := &v1.ServiceMeshMemberRoll{}
+		err := cl.Get(ctx, client.ObjectKey{Name: common.MemberRollName, Namespace: smcp.GetNamespace()}, smmr)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				// log error, but don't fail validation: we'll just assume that the control plane namespace is the only namespace for now
+				logger.Error(err, "failed to retrieve SMMR for SMCP")
+				smmr = nil
```

Comment:
I think that this assignment is right only when  `errors.IsNotFound(err)` is true. Otherwise we shouldn't assume anything, because of transient errors, like timeouts etc.

Reply:
again, same trade-off as above